src/main/java/apoc/load/Jdbc.java

```diff
@@ -169,36 +170,66 @@ public boolean hasNext() {
                 if (handleEndOfResults()) return null;
                 Map<String, Object> row = new LinkedHashMap<>(columns.length);
                 for (int col = 1; col < columns.length; col++) {
-                    row.put(columns[col], convert(rs.getObject(col)));
+                    int sqlType = rs.getMetaData().getColumnType(col);
+                    row.put(columns[col], convert(col, sqlType));
                 }
                 return row;
             } catch (SQLException e) {
                 throw new RuntimeException("Cannot execute read result-set.", e);
             }
         }
 
-        private Object convert(Object value) {
-            if (value instanceof UUID || value instanceof BigInteger || value instanceof BigDecimal) {
+        private Object convert(int col,int type) throws SQLException {
+            Object value = rs.getObject(col);
+            if(type == Types.TIMESTAMP ||type == Types.DATE){
+                return rs.getTimestamp(col).getTime();
+            }
+            if (type == Types.OTHER || type == Types.BIGINT || type == Types.DECIMAL) {
                 return value.toString();
             }
-            if (value instanceof java.util.Date) {
-                return ((java.util.Date) value).getTime();
+            if (type == Types.BLOB) {
+                Blob blob = ((Blob) value);
+                return getValue(blob.getBinaryStream());
+            }
+            if (type == Types.CLOB) {
+                Clob clob = ((Clob) value);
+                return getValue(clob.getAsciiStream());
+            }
+            if (type == Types.BINARY) {
+                byte[] bytes = (byte[]) value;
+                value = new String(bytes);
             }
+
             return value;
         }
 
+        private Object getValue(InputStream is){
+            try {
+                return new Scanner(is).next();
```

Comment:
Scanner is usually used for text only streams. And even then you have to set `\Z` as delimiter for EOF.
Probably better to use one of the included (apache) file-utils to read the full file.

Reply:
@jexp I would replace the scanner with IOUtils.toByteArray(is).